packages/core/src/repository/cache.repository.ts

```diff
@@ -16,22 +16,29 @@ export class CacheSyncOperation implements Operation {
 
 export interface ObjectValidator {
     isObjectValid<T>(object: T): boolean;
-    isArrayValid<T>(array: T[]): boolean;
 }
 
-export class DefaultObjectValidator implements ObjectValidator {
-    isObjectValid<T>(_object: T): boolean {
-        return true;
-    }
-    isArrayValid<T>(array: T[]): boolean {
-        if (array.length === 0) {
-            return false;
-        }
-        for (const element of array) {
-            if (!this.isObjectValid(element)) {
+export class ArrayValidator implements ObjectValidator {
+    constructor(private readonly validator: ObjectValidator) {}
+
+    public isObjectValid<T>(array: T): boolean {
+        if (Array.isArray(array)) {
+            if (array.length === 0) {
                 return false;
             }
+            for (const element of array) {
+                if (!this.validator.isObjectValid(element)) {
```

Comment:
Why not using `super.isObjectValid(element)` ?

Reply:
Because there's no super in the ArrayValidator. The usage will be the same as we do with mappers.
`new ArrayValidator(new MyCustomValidator())`